client/test/e2e/Menu.test.ts

```diff
@@ -2,100 +2,52 @@
-    await page.locator('div[role="button"]:has-text("Dashboard")').click();
-    await expect(page).toHaveURL('/dashboard');
-
-    await page.goto('/library');
-    await page.locator('div[role="button"]:has-text("Library")').click();
-    await expect(page).toHaveURL('/library');
-
-    await page.goto('/library');
-    await page.locator('div[role="button"]:has-text("Digital Twins")').click();
-    await expect(page).toHaveURL('/digitaltwins');
+interface TestLink {
+  text: string;
+  url: string;
+}
+const links: TestLink[] = [
+  { text: 'Dashboard', url: '/dashboard' },
+  { text: 'Library', url: '/library' },
+  { text: 'Digital Twins', url: '/digitaltwins' },
+  { text: 'Scenario Analysis', url: '/sanalysis' },
+  { text: 'History', url: '/history' },
+];
 
-    await page.goto('/library');
-    await page
-      .locator('div[role="button"]:has-text("Scenario Analysis")')
-      .click();
-    await expect(page).toHaveURL('/sanalysis');
-
-    await page.goto('/library');
-    await page.locator('div[role="button"]:has-text("History")').click();
-    await expect(page).toHaveURL('/history');
+test.describe('Header Contents and Navigation Links', () => {
+  links.forEach((link) => {
+    test(`Navigation to ${link.text} link on the dashboard page`, async ({
+      page,
+    }) => {
+      await page.goto(`${link.url}`);
+      await expect(page).toHaveURL(link.url);
+    });
   });
+});
 
-  test('Navigation Links on the Digital Twins page', async ({ page }) => {
-    await page.goto('/digitaltwins');
-    await page.locator('div[role="button"]:has-text("Dashboard")').click();
-    await expect(page).toHaveURL('/dashboard');
-
-    await page.goto('/digitaltwins');
-    await page.locator('div[role="button"]:has-text("Library")').click();
-    await expect(page).toHaveURL('/library');
-
-    await page.goto('/digitaltwins');
-    await page.locator('div[role="button"]:has-text("Digital Twins")').click();
-    await expect(page).toHaveURL('/digitaltwins');
-
-    await page.goto('/digitaltwins');
-    await page
-      .locator('div[role="button"]:has-text("Scenario Analysis")')
-      .click();
-    await expect(page).toHaveURL('/sanalysis');
-
-    await page.goto('/digitaltwins');
-    await page.locator('div[role="button"]:has-text("History")').click();
-    await expect(page).toHaveURL('/history');
+test.describe('Navigation from each link to other links', () => {
+  test.afterAll(async ({ browser, context }) => {
+    await context.close();
+    await browser.close();
   });
 
-  test('Navigation Links on the Scenario Analysis page', async ({ page }) => {
-    await page.goto('/sanalysis');
-    await page.locator('div[role="button"]:has-text("Dashboard")').click();
-    await expect(page).toHaveURL('/dashboard');
-
-    await page.goto('/sanalysis');
-    await page.locator('div[role="button"]:has-text("Library")').click();
-    await expect(page).toHaveURL('/library');
+  links.forEach((link) => {
```

Comment:
This is a strange convention. Usually we put code inside test.describe() function. Now we are iterating on the test() function itself. Can you move the link.forEach() into test.describe()?

Reply:
I'm pretty sure I read somewhere that it's also convention to loop outside of the test() function if needed (of course I failed to find where I read it). 

- In this case, we will have identical tests for each route. (Loop 1)
- Each route navigates to every link to check if it works. (Loop 2)

Looping twice inside the test gave us many issues.

For now I've taken it out of the PR, and will look into it more in detail for the next PR.